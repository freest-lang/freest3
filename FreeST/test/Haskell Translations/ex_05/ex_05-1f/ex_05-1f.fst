-- V exercise 1f

data CharList = End | List Char CharList

addBreakLine : CharList -> CharList
addBreakLine list = 
    case list of {
        End -> List ('\n') End,
        List c rest -> List c (addBreakLine rest)
    }

string : CharList
string = List 'a' (List 'b' (List 'c' (List 'd' (List 'e' End))))

main : CharList
main = addBreakLine string
-- expected result = List 'a' (List 'b' (List 'c' (List 'd' (List 'e' (List '\n' End)))))