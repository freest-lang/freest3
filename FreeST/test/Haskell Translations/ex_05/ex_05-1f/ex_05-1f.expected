List 'a' (List 'b' (List 'c' (List 'd' (List 'e' (List '\n' End)))))